plate_mapper/plate_mapper.py

```diff
@@ -68,22 +98,63 @@ def plate_mapper(input_f, barseq_f, output_f):
     barseq_f.close()
     print('  Done.')
     # Write output sequencing run file
+    samples = []
     print('Writing output mapping file...')
     for x in barseqs:
         # [ barcode sequence, linker primer sequence, primer plate #, well ID ]
         if x[2] in plates and x[3] in plates[x[2]]:
-            output_f.write('%s\t%s\t%s\n' % (plates[x[2]][x[3]], '\t'.join(x),
+            sample = plates[x[2]][x[3]]
+            output_f.write('%s\t%s\t%s\n' % (sample, '\t'.join(x),
                            '\t'.join(properties[x[2]])))
+            samples.append(sample)
         else:
             output_f.write('\t' + '\t'.join(x) + '\n')
     output_f.close()
-    print('  Done. Task completed.')
+    print('  Done.')
+    # Validate sample names
+    if names_f:
+        print('Validating sample names...')
+        samples = Counter(samples)
+        names = set()
+        for line in names_f:
+            l = line.rstrip().split('\t')
+            if l == [''] or l[0] == '':  # skip empty names
+                continue
+            names.add(l[0])  # keep first field as name
+        names_f.close()
+        warning = ''
+        if names:
+            sample_set = set(samples)
+            # samples in plate map but not in name list
+            novel = sample_set - names
+            # samples in name list but not in plate map
+            missing = names - sample_set
+            # samples that occur more than one times in plate map
+            repeated = set()
```

Comment:
This code does not loop the dict `names` (comparing to the original code), so I don't think it will work.